stdlib/2and3/cmath.pyi

```diff
@@ -14,29 +14,29 @@ if sys.version_info >= (3, 6):
 
 _C = Union[SupportsFloat, SupportsComplex]
 
-def acos(x: _C) -> complex: ...
-def acosh(x: _C) -> complex: ...
-def asin(x: _C) -> complex: ...
-def asinh(x: _C) -> complex: ...
-def atan(x: _C) -> complex: ...
-def atanh(x: _C) -> complex: ...
-def cos(x: _C) -> complex: ...
-def cosh(x: _C) -> complex: ...
-def exp(x: _C) -> complex: ...
+def acos(__z: _C) -> complex: ...
+def acosh(__z: _C) -> complex: ...
+def asin(__z: _C) -> complex: ...
+def asinh(__z: _C) -> complex: ...
+def atan(__z: _C) -> complex: ...
+def atanh(__z: _C) -> complex: ...
+def cos(__z: _C) -> complex: ...
+def cosh(__z: _C) -> complex: ...
+def exp(__z: _C) -> complex: ...
 if sys.version_info >= (3, 5):
     def isclose(a: _C, b: _C, *, rel_tol: SupportsFloat = ..., abs_tol: SupportsFloat = ...) -> bool: ...
-def isinf(z: _C) -> bool: ...
-def isnan(z: _C) -> bool: ...
-def log(x: _C, base: _C = ...) -> complex: ...
-def log10(x: _C) -> complex: ...
-def phase(z: _C) -> float: ...
-def polar(z: _C) -> Tuple[float, float]: ...
-def rect(r: float, phi: float) -> complex: ...
-def sin(x: _C) -> complex: ...
-def sinh(x: _C) -> complex: ...
-def sqrt(x: _C) -> complex: ...
-def tan(x: _C) -> complex: ...
-def tanh(x: _C) -> complex: ...
+def isinf(__z: _C) -> bool: ...
+def isnan(__z: _C) -> bool: ...
+def log(__x: _C, __y_obj: Optional[_C] = ...) -> complex: ...
```

Comment:
`None` is indeed the default for the second argument, but if you pass `None` it always throws

```
In [1339]: cmath.log(1, None)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/Volumes/EXTERNAL/Dropbox/code/taxonomy/taxonomy/db/constants.py in <module>()
----> 1 cmath.log(1, None)

TypeError: must be real number, not NoneType
```
So I think this should not be marked as `Optional`, and perhaps the documentation/signature for the function should be changedd to remove `None` as the claimed default.

Reply:
Oh that's tricky, will fix. Didn't realise that default value of None could result in different behaviour from passing None in (https://docs.python.org/3/howto/clinic.html#the-null-default-value)... That might slow down these stubtest changes.

Also I could not get this to return -1 https://github.com/hauntsaninja/cpython/blob/3.7/Objects/complexobject.c#L311

Reply:
It looks like it returns `-1j` if there is an error, but callers are supposed to call `PyErr_Occurred()` after a call to check for errors; that's what happens in the implementations of `cmath.log`.

Reply:
Wonder if it'd be reasonable to change https://github.com/hauntsaninja/cpython/blob/3.7/Modules/cmathmodule.c#L965 to check against Py_None as well.
```
~/dev/cpython 3.7 λ rg '(NULL.*Py_None|Py_None.*NULL)' | wc -l
     130
```

Reply:
I'd agree with that change, but I feel like at best such a change would only go into 3.9, so it doesn't really solve the issue in the stub.